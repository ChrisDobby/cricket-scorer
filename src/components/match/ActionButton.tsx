import * as React from 'react';

const buttonStyle: React.CSSProperties = {
    minWidth: '36px',
    margin: '2px',
};

export interface ActionButtonProps {
    caption: string;
    noBall: boolean;
    action: () => void;
}

export const ActionButton = ({ caption, noBall, action }: ActionButtonProps) => (
    <button
        className={`btn ${noBall ? 'btn-danger' : 'btn-success'}`}
        style={buttonStyle}
        onClick={action}
    >{caption}
    </button>
);
