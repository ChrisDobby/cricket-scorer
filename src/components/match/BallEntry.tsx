import * as React from 'react';
import * as globalStyles from '../styles';
import { Innings, Batter, Bowler, oversDescription, BattingInnings } from '../../domain';
import { EntryPanel, BallFunctions } from './EntryPanel';

const battingScoreStyle: React.CSSProperties = {
    display: 'inline',
    marginRight: '10px',
};

export interface BallEntryProps {
    innings: Innings;
    batter: Batter;
    bowler: Bowler;
    ballFunctions: BallFunctions;
}

export const BallEntry = ({ innings, batter, bowler, ballFunctions }: BallEntryProps) => (
    <div style={globalStyles.sectionContainer}>
        <div className="row" style={globalStyles.singleHeadingRow}>
            <h4>{`${bowler.name} to ${batter.name}`}</h4>
        </div>
        <div className="row">
            <div className="col-12 col-lg-6">
                <div className="row">
                    <div className="col-6">
                        <h5>{innings.battingTeam.name}</h5>
                    </div>
                    <div className="col-6">
                        <h5>{`${innings.score}-${innings.wickets}`}</h5>
                    </div>
                </div>
                <div className="row">
                    <div className="col-6">
                        <h5>Overs</h5>
                    </div>
                    <div className="col-6">
                        <h5>{oversDescription(innings.completedOvers, [])}</h5>
                    </div>
                </div>
                {innings.batting.batters.filter(batter => batter.innings && !batter.innings.wicket)
                    .map((batter, idx) => ({
                        name: batter.name,
                        innings: batter.innings as BattingInnings,
                        index: idx,
                    }))
                    .map(currentBatter => (
                        <div key={currentBatter.index} className="row">
                            <div className="col-6">
                                <h5>{currentBatter.name}</h5>
                            </div>
                            <div className="col-6">
                                <span>
                                    <h5 style={battingScoreStyle}>{`${currentBatter.innings.runs}` +
                                        `(${currentBatter.innings.ballsFaced})`}</h5>
                                    {currentBatter.name === batter.name && <i className="fa fa-circle" />}
                                </span>
                            </div>
                        </div>
                    ))}
                <div className="row">
                    <div className="col-6">
                        <h5>{bowler.name}</h5>
                    </div>
                    <div className="col-6">
                        <h5>
                            {`${oversDescription(bowler.completedOvers, [])}-` +
                                `${bowler.maidenOvers}-${bowler.runs}-${bowler.wickets}`}
                        </h5>
                    </div>
                </div>
            </div>
            <div className="col-12 col-lg-6">
                <EntryPanel ballFunctions={ballFunctions} />
            </div>
        </div>
    </div>
);
