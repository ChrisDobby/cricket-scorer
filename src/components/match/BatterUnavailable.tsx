import * as React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import { UnavailableReason, Innings } from '../../domain';

interface BatterUnavailableProps {
    reason: UnavailableReason;
    innings: Innings;
    battingPlayers: string[];
    update: (playerIndex: number, reason: UnavailableReason) => void;
    cancel: () => void;
}

export default (props: BatterUnavailableProps) => {
    const [playerIndex, setPlayerIndex] = React.useState(undefined as number | undefined);

    const batters = props.innings.batting.batters.filter(
        batter =>
            (props.reason === UnavailableReason.Absent && !batter.innings) ||
            (props.reason === UnavailableReason.Retired && batter.innings && !batter.innings.wicket),
    );
    const description = props.reason === UnavailableReason.Retired ? 'retired' : 'absent';

    return (
        <div>
            <Dialog open={true} aria-labelledby="update-innings-overs-title">
                <DialogTitle id="update-innings-overs-title">{`Mark one of the batting side as ${description}`}</DialogTitle>
                <DialogContent>
                    <DialogContentText>{`Select the ${description} player`}</DialogContentText>
                </DialogContent>
                <DialogContent>
                    {batters.length > 0 && (
                        <Select value={playerIndex} onChange={ev => setPlayerIndex(Number(ev.target.value))} fullWidth>
                            {batters.map(batter => (
                                <MenuItem key={batter.playerIndex} value={batter.playerIndex}>
                                    {props.battingPlayers[batter.playerIndex]}
                                </MenuItem>
                            ))}
                        </Select>
                    )}
                    {batters.length === 0 && (
                        <DialogContentText>{`No players can be marked as ${description}`}</DialogContentText>
                    )}
                </DialogContent>
                {batters.length > 0 && (
                    <DialogActions>
                        <Button
                            onClick={() => props.update(playerIndex as number, props.reason)}
                            color="primary"
                            autoFocus
                            disabled={typeof playerIndex === 'undefined'}
                        >
                            OK
                        </Button>
                        <Button onClick={props.cancel} color="primary">
                            Cancel
                        </Button>
                    </DialogActions>
                )}
                {batters.length === 0 && (
                    <DialogActions>
                        <Button onClick={props.cancel} color="primary" autoFocus>
                            OK
                        </Button>
                    </DialogActions>
                )}
            </Dialog>
        </div>
    );
};
