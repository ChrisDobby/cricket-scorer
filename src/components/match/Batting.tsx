import * as React from 'react';
import { Extras } from './Extras';
import { Batting as InningsBatting, BattingInnings, howOutDescription } from '../../domain';
import * as styles from './styles';

const battingContainer: React.CSSProperties = {
    width: '100%',
    padding: '20px',
    marginBottom: '20px',
};

const howOut = (innings?: BattingInnings): string => {
    if (!innings) { return ''; }

    return howOutDescription(innings.wicket);
};

interface InningsItemProps { innings?: BattingInnings; }
const Howout = (props: InningsItemProps) => (
    <div className="col-md-4">{howOut(props.innings)}</div>
);

interface CellItemProps {
    property: string;
    style: React.CSSProperties;
}
const CellItem = ({ property, style }: CellItemProps) => (
    <div className="col-md-1" style={style}>{property}</div>
);

export interface BattingProps { batting: InningsBatting; }

export const Batting = ({ batting }: BattingProps) => (
    <div className="col-xl-8 col-lg-12">
        <div style={battingContainer}>
            <div className="row" style={styles.headingRow}>
                <div className="col-md-7"><h6>Batsman</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>Runs</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>Balls</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>Mins</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>4s</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>6s</h6></div>
            </div>    
            {batting.batters.map(batter =>
                <div className="row" style={styles.itemRow} key={batter.position}>
                    <div className="col-md-3">{batter.name}</div>
                    <Howout innings={batter.innings} />
                    <CellItem property={batter.innings ? batter.innings.runs.toString() : ''} style={styles.runsCell} />
                    <CellItem
                        property={batter.innings ? batter.innings.ballsFaced.toString() : ''}
                        style={styles.numberCell}
                    />
                    <CellItem property={batter.innings ? '0' : ''} style={styles.numberCell} />
                    <CellItem
                        property={batter.innings ? batter.innings.fours.toString() : ''}
                        style={styles.numberCell}
                    />
                    <CellItem
                        property={batter.innings ? batter.innings.sixes.toString() : ''}
                        style={styles.numberCell}
                    />
                </div>)}
            <Extras extras={batting.extras} />
        </div>
    </div>
);
