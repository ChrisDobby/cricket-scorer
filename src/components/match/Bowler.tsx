import * as React from 'react';
import * as styles from './styles';
import * as globalStyles from '../styles';

const indicatorStyle: React.CSSProperties = {
    ...globalStyles.primaryColour,
    float: 'left',
    width: '30px',
    textAlign: 'center',
};

const nonSelectableStyle: React.CSSProperties = {
    color: 'rgb(233, 236, 239)',
};

interface SelectionIndicatorProps {
    playerIndex: number;
    selected: boolean;
}

const SelectionIndicator = ({ playerIndex, selected }: SelectionIndicatorProps) => {
    return (
        <div style={indicatorStyle}>
            {selected && <i className="fa fa-circle" />}
        </div>
    );
};

export interface BowlerProps {
    index: number;
    selected: boolean;
    name: string;
    allowed: boolean;
    selectBowler: (bowler: number) => void;
}

export const Bowler = ({ index, selected, name, allowed, selectBowler }: BowlerProps) => (
    <div
        className="row bowler-row"
        style={allowed ? styles.selectablePlayerStyle : nonSelectableStyle}
        onClick={() => { if (allowed) { selectBowler(index); } }}
    >
        <SelectionIndicator
            playerIndex={index}
            selected={selected}
        />
        <h6>{name}</h6>
    </div>
);
