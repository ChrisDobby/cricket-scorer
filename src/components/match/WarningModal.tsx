import * as React from 'react';

const fullWindowStyle: React.CSSProperties = {
    left: 0,
    top: 0,
    right: 0,
    bottom: 0,
    position: 'fixed',
};

const modalStyle: React.CSSProperties = {
    ...fullWindowStyle,
    backgroundColor: '#000000',
    opacity: 0.6,
    zIndex: 9000,
};

const warningStyle: React.CSSProperties = {
    ...fullWindowStyle,
    zIndex: 9001,
    padding: '20px',
};

const alertStyle: React.CSSProperties = {
    textAlign: 'center',
};

const buttonStyle: React.CSSProperties = {
    marginLeft: '10px',
    marginRight: '10px',
};

export enum WarningType {
    OverNotCompleteWarning,
    AllRunFourWarning,
    AllRunSixWarning,
}

export interface WarningModalProps {
    warningType: WarningType;
    onYes: () => void;
    onNo: () => void;
}

const allRunWarningText = (runs: string) =>
    `You have entered a ${runs} as all run rather than a boundary.  Is this correct?`;

const warningText = (warningType: WarningType) => {
    switch (warningType) {
    case WarningType.OverNotCompleteWarning:
        return 'There has not been six legal deliveries in this over.  Do you still want to complete it?';
    case WarningType.AllRunFourWarning:
        return allRunWarningText('four');
    case WarningType.AllRunSixWarning:
        return allRunWarningText('six');
    default:
        return '';
    }
};

export const WarningModal = ({ warningType, onYes, onNo }: WarningModalProps) => (
    <div>
        <div style={modalStyle} />
        <div className="row" style={warningStyle}>
            <div className="col-12" style={alertStyle}>
                <div className="alert alert-danger" role="alert">
                    {warningText(warningType)}
                    <div>
                        <button
                            className="btn btn-danger"
                            style={buttonStyle}
                            onClick={onYes}
                        >Yes
                        </button>
                        <button
                            className="btn btn-default"
                            style={buttonStyle}
                            onClick={onNo}
                        >No
                        </button>
                    </div>
                </div>
            </div>
        </div>
    </div>
);
