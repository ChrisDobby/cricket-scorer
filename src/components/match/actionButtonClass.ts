import { DeliveryOutcome } from '../../domain';

const actionButtonClass = (invalidClassName: string) => (deliveryOutcome: DeliveryOutcome) =>
    deliveryOutcome === DeliveryOutcome.Noball || deliveryOutcome === DeliveryOutcome.Wide
        ? invalidClassName
        : undefined;

export default actionButtonClass('btn-danger');
