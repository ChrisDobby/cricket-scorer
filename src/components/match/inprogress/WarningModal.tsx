import * as React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';

export enum WarningType {
    AllRunFourWarning,
    AllRunSixWarning,
}

export interface WarningModalProps {
    warningType: WarningType;
    onYes: () => void;
    onNo: () => void;
}

const allRunWarningText = (runs: string) =>
    `You have entered a ${runs} as all run rather than a boundary.  Is this correct?`;

const warningText = (warningType: WarningType) => {
    switch (warningType) {
        case WarningType.AllRunFourWarning:
            return allRunWarningText('four');
        case WarningType.AllRunSixWarning:
            return allRunWarningText('six');
    }
};

export const WarningModal = ({ warningType, onYes, onNo }: WarningModalProps) => (
    <div>
        <Dialog open={true} aria-labelledby="all-run-warning-title">
            <DialogTitle id="all-run-warning-title">All run</DialogTitle>
            <DialogContent>
                <DialogContentText>{warningText(warningType)}</DialogContentText>
            </DialogContent>
            <DialogActions>
                <Button onClick={onYes} color="primary" autoFocus>
                    Yes
                </Button>
                <Button onClick={onNo} color="primary">
                    No
                </Button>
            </DialogActions>
        </Dialog>
    </div>
);
