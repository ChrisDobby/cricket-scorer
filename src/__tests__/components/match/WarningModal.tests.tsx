import * as React from 'react';
import { shallow } from 'enzyme';
import * as ReactTestRenderer from 'react-test-renderer';
import { WarningModal, WarningType } from '../../../components/match/WarningModal';

jest.mock('../../../components/WithModal', () => ({
    __esModule: true,
    namedExport: jest.fn(),
    default: jest.fn(Component => Component),
}));

describe('WarningModal', () => {
    it('should call yes function when yes button clicked', () => {
        const yes = jest.fn();
        const warning = shallow(
            <WarningModal
                warningType={WarningType.OverNotCompleteWarning}
                onYes={yes}
                onNo={jest.fn()}
            />);

        warning.find('.btn-danger').at(0).simulate('click');

        expect(yes).toHaveBeenCalled();
    });

    it('should call no function when no button clicked', () => {
        const no = jest.fn();
        const warning = shallow(
            <WarningModal
                warningType={WarningType.OverNotCompleteWarning}
                onYes={jest.fn()}
                onNo={no}
            />);

        warning.find('.btn-default').at(0).simulate('click');

        expect(no).toHaveBeenCalled();
    });

    it('should render correctly for not complete over warning', () => {
        const warning = ReactTestRenderer.create(
            <WarningModal
                warningType={WarningType.OverNotCompleteWarning}
                onYes={jest.fn()}
                onNo={jest.fn()}
            />);

        expect(warning.toJSON()).toMatchSnapshot();
    });

    it('should render correctly for all run 4 warning', () => {
        const warning = ReactTestRenderer.create(
            <WarningModal
                warningType={WarningType.AllRunFourWarning}
                onYes={jest.fn()}
                onNo={jest.fn()}
            />);

        expect(warning.toJSON()).toMatchSnapshot();
    });

    it('should render correctly for all run 6 warning', () => {
        const warning = ReactTestRenderer.create(
            <WarningModal
                warningType={WarningType.AllRunSixWarning}
                onYes={jest.fn()}
                onNo={jest.fn()}
            />);

        expect(warning.toJSON()).toMatchSnapshot();
    });
});
