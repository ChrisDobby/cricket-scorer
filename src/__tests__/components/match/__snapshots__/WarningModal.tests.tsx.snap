// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningModal should render correctly for all run 4 warning 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#000000",
        "bottom": 0,
        "left": 0,
        "opacity": 0.6,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 9000,
      }
    }
  />
  <div
    className="row"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "padding": "20px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 9001,
      }
    }
  >
    <div
      className="col-12"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        You have entered a four as all run rather than a boundary.  Is this correct?
        <div>
          <button
            className="btn btn-danger"
            onClick={[MockFunction]}
            style={
              Object {
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
          >
            Yes
          </button>
          <button
            className="btn btn-default"
            onClick={[MockFunction]}
            style={
              Object {
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
          >
            No
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WarningModal should render correctly for all run 6 warning 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#000000",
        "bottom": 0,
        "left": 0,
        "opacity": 0.6,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 9000,
      }
    }
  />
  <div
    className="row"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "padding": "20px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 9001,
      }
    }
  >
    <div
      className="col-12"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        You have entered a six as all run rather than a boundary.  Is this correct?
        <div>
          <button
            className="btn btn-danger"
            onClick={[MockFunction]}
            style={
              Object {
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
          >
            Yes
          </button>
          <button
            className="btn btn-default"
            onClick={[MockFunction]}
            style={
              Object {
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
          >
            No
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WarningModal should render correctly for not complete over warning 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#000000",
        "bottom": 0,
        "left": 0,
        "opacity": 0.6,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 9000,
      }
    }
  />
  <div
    className="row"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "padding": "20px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 9001,
      }
    }
  >
    <div
      className="col-12"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        There has not been six legal deliveries in this over.  Do you still want to complete it?
        <div>
          <button
            className="btn btn-danger"
            onClick={[MockFunction]}
            style={
              Object {
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
          >
            Yes
          </button>
          <button
            className="btn btn-default"
            onClick={[MockFunction]}
            style={
              Object {
                "marginLeft": "10px",
                "marginRight": "10px",
              }
            }
          >
            No
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
